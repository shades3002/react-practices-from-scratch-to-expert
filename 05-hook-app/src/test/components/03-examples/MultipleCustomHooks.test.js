import { shallow } from 'enzyme';
import React from 'react';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('test in MultipleCustomHooks', () => {

    beforeEach( () => {

        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });

    });


    test('should show correctly', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks />);
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should show the info', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Carlos',
                quote: 'hi'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks />);

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-4').text().trim() ).toBe( 'hi' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Carlos' );

    });
    
});
