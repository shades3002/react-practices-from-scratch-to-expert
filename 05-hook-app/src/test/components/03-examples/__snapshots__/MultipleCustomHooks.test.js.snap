// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in MultipleCustomHooks should show correctly 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
